app/WebParsing/UtsgJsonParser.hs

```diff
@@ -47,44 +47,37 @@ insertAllCourses org = do
         -- only sections are currently stored here.
         (_, sections) = unzip courseData
         meetings = concat sections
-    mapM_ insertCourse meetings
+    mapM_ insertMeeting meetings
 
-getCourseKey :: T.Text -> SqlPersistM (Maybe (Key Courses))
-getCourseKey code = do
-    keyListCourse :: [Key Courses] <- selectKeysList [ CoursesCode ==. code ] []
-    return $ case keyListCourse of
-        [] -> Nothing
-        _ -> Just (head keyListCourse)
-
-insertCourse :: Meeting -> SqlPersistM ()
-insertCourse meet = do
-    courseKey <- getCourseKey (meetingCode meet)
+-- | Store a meeting's data and times.
+insertMeeting :: MeetTimes -> SqlPersistM ()
+insertMeeting meet = do
+    -- Check that the meeting belongs to a course that exists
+    let code = meetingCode $ meetingData meet
+    courseKey <- selectFirst [ CoursesCode ==. code ] []
     case courseKey of
-        Just _ -> insert_ meet
+        Just _ -> do
+          meetingKey <- insert $ meetingData meet
+          mapM_ (\t -> insert_ $ t {timesMeeting = Just meetingKey}) $ timesData meet
         Nothing -> return ()
 
-newtype DB = DB { dbData :: (Courses, [Meeting]) }
+parseMeetingTimes:: T.Text -> T.Text -> Meeting -> [Times] -> MeetTimes
+parseMeetingTimes code session meetTimes allTimes =
+    MeetTimes {meetingData = meetTimes {meetingCode = code, meetingSession = session}, timesData = allTimes }
+
+newtype DB = DB { dbData :: (Courses, [MeetTimes]) }
   deriving Show
 
 instance FromJSON DB where
     parseJSON (Object o) = do
       course <- parseJSON (Object o)
       session :: T.Text <- o .:? "section" .!= "F"
-      meetingMap :: HM.HashMap T.Text Meetings <- o .:? "meetings" .!= HM.empty
-      let meetings = map (setCode (coursesCode course) session . meeting) (HM.elems meetingMap)
+      meetingTimesMap :: HM.HashMap T.Text MeetTimes <- o .:? "meetings" .!= HM.empty
+      let allMeetingsTimes = map (\meetTime -> parseMeetingTimes (coursesCode course) session (meetingData meetTime) (timesData meetTime)) (HM.elems meetingTimesMap)
```

Comment:
This code does things in a roundabout way and can be simplified. Don't include a helper here; instead, just map something like `(\m -> m { meetingData = map (\d -> d { meetingCode = code, meetingSession = session }) $ meetingData m).